### ISSUE TITLE
[serve]  ValueError: Keys not found

### ISSUE THREAD

**Issue by crypdick** [ray-project/ray#15648]
### What is the problem?
While running a Serve batch predict job, this error is printed thousands of times:

```
2021-04-28 14:27:18,061 ERROR worker.py:1056 -- Possible unhandled error from worker: ray::ServeController.listen_for_change() (pid=5229, ip=172.31.4.63)
  File "python/ray/_raylet.pyx", line 505, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 446, in ray._raylet.execute_task.function_executor
  File "python/ray/_raylet.pyx", line 1632, in ray._raylet.CoreWorker.run_async_func_in_event_loop
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/home/ubuntu/seeweed/venv/lib/python3.8/site-packages/ray/serve/controller.py", line 99, in listen_for_change
    return await (
  File "/home/ubuntu/seeweed/venv/lib/python3.8/site-packages/ray/serve/long_poll.py", line 162, in listen_for_change
    raise ValueError(f"Keys not found: {nonexistent_keys}.")
ValueError: Keys not found: {(LongPollNamespace.TRAFFIC_POLICIES, 'predictor')}.
```

Ray version 1.3.0
Python 3.8.5
Ubuntu 20.10

Unfortunately, I do not know how to reproduce this in a minimal example. cc @simon-mo

**Comment by simon-mo** [comment]
Thanks @crypdick, I will look into this.

**Comment by edoakes** [comment]
@simon-mo were you ever able to repro this?

**Comment by crypdick** [comment]
@edoakes if not, I'm happy to do a pair debugging session. I see this issue even with smoke testing on 10 images.

**Comment by simon-mo** [comment]
@crypdick sorry about the delay, can you try to reproduce this with 1.4.0?

**Comment by crypdick** [comment]
@simon-mo sure, we're still on 1.3.0

**Comment by crypdick** [comment]
@simon-mo @edoakes seems fixed in 1.4.0, although I'm seeing other verbose errors. That's for a separate issue.
